adapter/repository/in-memory-simple/src/main/java/org/mobilitydata/gtfsvalidator/db/InMemoryGtfsDataRepository.java

```diff
@@ -98,4 +100,47 @@ public Route addRoute(@NotNull final Route newRoute) throws IllegalArgumentExcep
     public Route getRouteById(final String routeId) {
         return routeCollection.get(routeId);
     }
+
+    /**
+     * Add a Transfer representing a row from transfers.txt to this. Return the entity added to the repository if the
+     * uniqueness constraint of route based on composite key from_stop_id and to_stop_id is respected, if this
+     * requirement is not met, returns null.
+     *
+     * @param newTransfer the internal representation of a row from transfers.txt to be added to the repository.
+     * @return the entity added to the repository if the uniqueness constraint of route based on composite key
+     * from_stop_id and to_stop_id is respected, if this requirement is not met, returns null.
+     */
+    @Override
+    public Transfer addTransfer(final Transfer newTransfer) throws IllegalArgumentException {
+        if (newTransfer != null) {
+            if ((transferCollection.containsKey(newTransfer.getFromStopId())) &&
+                    (transferCollection.get(newTransfer.getFromStopId()).containsKey(newTransfer.getToStopId()))) {
+                return null;
+            } else {
+                String fromStopId = newTransfer.getFromStopId();
```

Comment:
There needs to be a check if `transferCollection.get(fromStopId)` exists (and if so, adding the `toStopId` to that existing `Map`) before doing `transferCollection.put(fromStopId, innerMap)`. Otherwise you're always overwriting the existing map for the given `fromStopId`, and the max size of the inner map will always be 1.

Reply:
Will modify